File: 020/Haskell/Problem020.hs
module Problem020 (sumOfDigitsOfFactorial, sumOfDigits) where

import Numbers

sumOfDigitsOfFactorial :: Integer -> Integer
sumOfDigitsOfFactorial = sumOfDigits . factorial

sumOfDigits n
  | n < 10    = n
  | otherwise = (n `mod` 10) + sumOfDigits (n `div` 10)

File: 020/Python/problem020.py
import numbers

def sum_of_digits_of_factorial(n):
    return sum_of_digits(numbers.factorial(n))
    
def sum_of_digits(n):
    if n < 10:
        return n
    else:
        return (n % 10) + sum_of_digits(n // 10)
